win32ss/user/ntuser/winpos.c

```diff
@@ -3709,35 +3708,43 @@ NtUserSetWindowPlacement(HWND hWnd,
    TRACE("Enter NtUserSetWindowPlacement\n");
    UserEnterExclusive();
 
-   if (!(Wnd = UserGetWindowObject(hWnd)) ||
-        UserIsDesktopWindow(Wnd) || UserIsMessageWindow(Wnd))
-   {
-      goto Exit; // Return FALSE
-   }
+    _SEH2_TRY
+    {
+        ProbeForRead(lpwndpl, sizeof(WINDOWPLACEMENT), 1);
+        Safepl = *lpwndpl;
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        SetLastNtError(_SEH2_GetExceptionCode());
+        _SEH2_YIELD(goto Exit); // Return FALSE
+    }
+    _SEH2_END
 
-   _SEH2_TRY
-   {
-      ProbeForRead(lpwndpl, sizeof(WINDOWPLACEMENT), 1);
-      RtlCopyMemory(&Safepl, lpwndpl, sizeof(WINDOWPLACEMENT));
-   }
-   _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
-   {
-      SetLastNtError(_SEH2_GetExceptionCode());
-      _SEH2_YIELD(goto Exit); // Return FALSE
-   }
-   _SEH2_END
+    /* Backwards-compatibility: Win 3.x doesn't check the length */
+    if (LOWORD(gptiCurrent->dwExpWinVer) < WINVER_WINNT4)
+        Safepl.length = sizeof(WINDOWPLACEMENT);
```

Comment:
Note: the sizeof could be `sizeof(Safepl)`

Reply:
```suggestion
        Safepl.length = sizeof(Safepl);
```